### DISCUSSION TITLE
is there a way to update the page properties contents of one database, using only one request to notion server ?

### DISCUSSION THREAD

**Discussion by tyehuoPHD** [jheddings/notional#131] [Ideas]
# `notion.pages.update` could update one page's property contents in one database.  but my database has over 1000 rows.

Say, I have a database having 1000 rows(pages). Is there a way to update the all the pages, using only one request to notion server ?
I have read the notion api documentations. It seems unlikely.

**Comment by jheddings** [comment]
Your suspicion is correct...  With Notion, you will need to loop over the pages in the database and make the change to each page.

Using Notional, your script might look something like this:

```python
notion = notional.connect(auth=my_auth_token)
query = EndpointIterator(notion.databases().query)

# loop over all pages in the database
for page in query(database_id=my_database_id):

    # set the property to the desired value using the appropriate type
    notion.pages.update(page, Complete=types.Checkbox[True])
```

**Reply by tyehuoPHD** [reply]
thank you. codes in example folder are very helpful.
